Update If-match header (#114)
elasticpath/elasticpath-dev: guides/How-To/commerce-extensions/create-a-multilocation-inventories-resource.md

        "field_type": "string",
        "validation": {
          "string": {

>             "regex": "^(Eiffel Tower|Graceland|Hamptons)$"

Reviewer: ```suggestion
            "regex": "^(Paris|London|New York)$"
```

      }
    }
```

> When using Custom API Entries, if multiple independent clients update the same resource, you should have them use the `If-Match` header to ensure data consistency and prevent lost updates and inconsistencies in inventory amounts. For example, if two users perform the same operation simultaneously, the `If-Match` header ensures that only the latest version of the inventory is updated. This prevents scenarios where concurrent updates could lead to incorrect inventory counts. This header works by comparing the provided ETag value with the current ETag value of the resource. If the resource hasn't changed since you last read it, the ETag will not change, ensuring the update is safe.

Reviewer: ```suggestion
When using Custom API Entries, if multiple independent clients update the same resource, you should have them use the `If-Match` header to prevent lost updates and other data consistency issues in the inventory amounts. For example, if two users see an amount of 3 and allocate 1, then both would update the amount to 2. The `If-Match` header ensures that only one of these requests succeeds. This header works by comparing the provided ETag value with the current ETag value of the resource. If the resource hasn't changed since you last read it, the ETag will not change, ensuring the update is safe. 
```

> To update a resource, include the `If-Match` header in your request and set its value to the ETag. For more information, see [update a custom entry](https://elasticpath.dev/docs/api/commerceextensions/update-a-custom-entry). If the value of the header matches the current ETag, the request completes successfully. If not, `HTTP 412 Precondition Failed` is returned.

Reviewer: ```suggestion
To update a resource, include the `If-Match` header in your request and set its value to the ETag. For more information, see [update a custom entry](https://elasticpath.dev/docs/api/commerceextensions/update-a-custom-entry). If the value of the header matches the current ETag, the request completes successfully. If not, `HTTP 412 Precondition Failed` is returned. If a client receives an `HTTP 412`, they should reread the resource to get the latest version, ensure the update is safe (i.e., check that the amount is greater than zero), and try again.
```

The following example shows how to update a custom API entry for location inventories:

```bash
curl -X PUT "https://useast.api.elasticpath.com/v2/extensions/location-inventories/:customApiEntryId" \
     -H "Authorization: XXXX" \
     -H "Content-Type: application/json" \
     -H "If-Match: W/\"{etag_id}\"" \
     -d $ {
      "data": {
        "type": "location_inventory_ext",
        "amount": 3
      }
    }
```

## Example filters for Common Queries

A customer is on a product page and lives near the Hamptons store and wants to know if this product in this color is available at their local store. Filtering on `slug` and `location-name` will retrieve the record for that location for that SKU.
```sh

> curl -X GET "https://useast.api.elasticpath.com/v2/extensions/location-inventories?filter=eq(slug,ABC123):eq(location-name,Hamptions)" \

Reviewer: ```suggestion
curl -X GET "https://useast.api.elasticpath.com/v2/extensions/location-inventories?filter=eq(slug,ABC123):eq(location-name,New York)" \
```

      }
    }
```

> With Custom API Entries, you can set `If-Match` header to ensure data consistency when updating resources, such as inventory. This header works by comparing the provided ETag value with the current ETag value of the resource. This ETag is a unique identifier representing the current version of the resource.

Reviewer: ```suggestion
With Custom API Entries, if multiple independent clients might update the same resource you should have them use the `If-Match` header to prevent lost updates, and inconsistency in the amount of inventory. This header works by comparing the provided ETag value with the current ETag value of the resource. If the resource hasn't changed since the time you read it, the ETag will not change and the update is safe.
```
